### PR TITLE
Python3 conversion

### PR DESCRIPTION
**Pull request by klauern**
# Description

This is a small automation to run `pyupgrade`
(https://github.com/asottile/pyupgrade) against the ScoutSuite repository, using
the `--py36-plus` flag, which updates as much code as possible, including using f-strings, but migrates a lot of formatting from the 2.7+ versions to more native 3.x styles.

There are a few types of changes, but I see that almost all of them fall into
these categories:

* `super()` calls - https://github.com/asottile/pyupgrade#super-calls
* "new style" classes - https://github.com/asottile/pyupgrade#new-style-classes
* Unicode literals - https://github.com/asottile/pyupgrade#unicode-literals
* various `__future__` import removals https://github.com/asottile/pyupgrade#__future__-import-removal
* Python 2 format specifiers https://github.com/asottile/pyupgrade#python27-format-specifiers
* f-strings, lots and lots of f-strings

## Type of change

Select the relevant option(s):

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (optional)
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: ScoutSuite/providers/aws/facade/ec2.py [hunk 1]
```diff
@@ -102,12 +102,19 @@ async def get_network_interfaces(self, region: str, vpc: str):
     async def get_volumes(self, region: str):
         try:
             volumes = await AWSFacadeUtils.get_all_pages('ec2', region, self.session, 'describe_volumes', 'Volumes')
-            await get_and_set_concurrently([self._get_and_set_key_manager], volumes, region=region)
+            await get_and_set_concurrently([self._get_and_set_key_manager, self._get_and_set_volume_tags], volumes, region=region)
```
Comment (alessandrogonzalez) [inline ScoutSuite/providers/aws/facade/ec2.py:105]: Some changes in this file (like this example) don't look like syntax changes. Is this PR mixing some content from other (unrelated) PRs?

File: ScoutSuite/providers/aws/resources/ec2/ami.py [hunk 2]
```diff
@@ -14,6 +14,9 @@ async def fetch_all(self):
             self[name] = resource
 
     def _parse_image(self, raw_image):
-        raw_image['id'] = raw_image.get('ImageId')
-        raw_image['name'] = raw_image.get('Name')
+        raw_image['id'] = raw_image['ImageId']
+        raw_image['name'] = raw_image['Name']
+        if 'Tags' in raw_image:
```
Comment (alessandrogonzalez) [inline ScoutSuite/providers/aws/resources/ec2/ami.py:19]: The same as my previous comment on lines 19 and 20 of this change

Comment (klauern) [reply ScoutSuite/providers/aws/resources/ec2/ami.py:19]: Ah, yeah, I will remove this one.  Sorry about that.

File: ScoutSuite/providers/aws/facade/ec2.py (hunk 1 above)
Comment (klauern) [reply ScoutSuite/providers/aws/facade/ec2.py:105]: This is also part of my messup.  I'm working off of a fork, so I will get these two changes reverted.

File: ScoutSuite/providers/aws/resources/ec2/ami.py (hunk 2 above)
Comment (alessandrogonzalez) [reply ScoutSuite/providers/aws/resources/ec2/ami.py]: Sorry, I was still reviewing the changes. There are still changes related to tags in the following files:

- ScoutSuite/providers/aws/resources/ec2/instances.py
- ScoutSuite/providers/aws/resources/ec2/snapshots.py
- ScoutSuite/providers/aws/resources/ec2/volumes.py

**Comment by j4v** [comment]
> using the --py3-plus flag, which retains the 3.5 limits from using f-strings

Also happy for this PR to remove support for 3.5.

**Comment by j4v** [comment]
I'm closing the PR as this has been merged into https://github.com/nccgroup/ScoutSuite/tree/release/5.10 and will be merged into `master` as part of v5.10 (https://github.com/nccgroup/ScoutSuite/pull/804).

**Pull request closed without merging**